services/stickers/api.go

```
@@ -0,0 +1,648 @@
+)
+
+func NewAPI(ctx context.Context, appDB *sql.DB, rpcClient *rpc.Client, accountsManager *account.GethManager, rpcFiltersSrvc *rpcfilters.Service, config *params.NodeConfig) *API {
+	return &API{
+		contractMaker: &contractMaker{
+			RPCClient: rpcClient,
+		},
+		accountsManager: accountsManager,
+		accountsDB:      accounts.NewDB(appDB),
+		rpcFiltersSrvc:  rpcFiltersSrvc,
+		config:          config,
+		ctx:             ctx,
+		client: &http.Client{
+			Timeout: time.Second * 5,
+		},
+	}
+}
+
+// ConnectionType constants
+type stickerStatus int
+
+const (
+	statusInstalled stickerStatus = iota
+	statusPurchased
+	// statusPending
+)
+
+type API struct {
+	contractMaker   *contractMaker
+	accountsManager *account.GethManager
+	accountsDB      *accounts.Database
+	rpcFiltersSrvc  *rpcfilters.Service
+	config          *params.NodeConfig
+	ctx             context.Context
+	client          *http.Client
+}
+
+type Sticker struct {
+	PackID *bigint.BigInt `json:"packID"`
+	URL    string         `json:"url"`
+	Hash   string         `json:"hash"`
+}
+
+type StickerPack struct {
+	ID        *bigint.BigInt `json:"id"`
+	Name      string         `json:"name"`
+	Author    string         `json:"author"`
+	Owner     common.Address `json:"owner"`
+	Price     *bigint.BigInt `json:"price"`
+	Preview   string         `json:"preview"`
+	Thumbnail string         `json:"thumbnail"`
+	Stickers  []Sticker      `json:"stickers"`
+
+	Status stickerStatus `json:"status"`
+}
+
+type ednSticker struct {
+	Hash string
+}
+
+type ednStickerPack struct {
+	Name      string
+	Author    string
+	Thumbnail string
+	Preview   string
+	Stickers  []ednSticker
+}
+type ednStickerPackInfo struct {
+	Meta ednStickerPack
+}
+
+func (api *API) Uninstall(packID uint64) error {
+	installedPacks, err := api.Installed()
+	if err != nil {
+		return err
+	}
+
+	if _, exists := installedPacks[uint(packID)]; !exists {
+		return nil // Pack is not installed
```

shall we return an error here?